Invalid types for IntroStep (TypeScript)

## Description

IntroStep types do not match documentation.

## Expected Behavior

I expect to be able to add steps via the API specifying `element`, `intro`, and `position` properties as per the documentation: https://introjs.com/docs/intro/api#introjsaddstepssteps

## Actual Behavior

This results in a type error because the [IntroStep](https://github.com/usablica/intro.js/blob/master/src/core/steps.ts#L32) type includes `step`, `title`, and `scrollTo` as required properties.

```
error TS2345: Argument of type '{ element: string; intro: string; position: string; }' is not assignable to parameter of type 'IntroStep'.
  Type '{ element: string; intro: string; position: string; }' is missing the following properties from type 'IntroStep': step, title, scrollTo
```

When I ignore the type error it seems to work, so it should just be a matter of updating the types.

--- a/src/core/DOMEvent.ts
+++ b/src/core/DOMEvent.ts
@@ -1,4 +1,4 @@
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 import stamp from "../util/stamp";
 
 /**

--- a/src/core/addOverlayLayer.ts
+++ b/src/core/addOverlayLayer.ts
@@ -1,7 +1,7 @@
 import exitIntro from "./exitIntro";
 import createElement from "../util/createElement";
 import setStyle from "../util/setStyle";
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 
 /**
  * Add overlay layer to the page

--- a/src/core/dontShowAgain.ts
+++ b/src/core/dontShowAgain.ts
@@ -1,4 +1,4 @@
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 import { deleteCookie, getCookie, setCookie } from "../util/cookie";
 
 const dontShowAgainCookieValue = "true";

--- a/src/core/exitIntro.ts
+++ b/src/core/exitIntro.ts
@@ -3,7 +3,7 @@ import onKeyDown from "./onKeyDown";
 import onResize from "./onResize";
 import removeShowElement from "./removeShowElement";
 import removeChild from "../util/removeChild";
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 import isFunction from "../util/isFunction";
 
 /**

--- a/src/core/fetchIntroSteps.ts
+++ b/src/core/fetchIntroSteps.ts
@@ -1,4 +1,4 @@
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 import cloneObject from "../util/cloneObject";
 import createElement from "../util/createElement";
 import { IntroStep, ScrollTo, TooltipPosition } from "./steps";

--- a/src/core/hint.ts
+++ b/src/core/hint.ts
@@ -10,7 +10,7 @@ import placeTooltip from "./placeTooltip";
 import createElement from "../util/createElement";
 import debounce from "../util/debounce";
 import { HintPosition, HintStep, TooltipPosition } from "./steps";
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 import isFunction from "../util/isFunction";
 
 /**

--- a/src/core/introForElement.ts
+++ b/src/core/introForElement.ts
@@ -4,7 +4,7 @@ import { nextStep } from "./steps";
 import onKeyDown from "./onKeyDown";
 import onResize from "./onResize";
 import fetchIntroSteps from "./fetchIntroSteps";
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 import isFunction from "../util/isFunction";
 
 /**

--- a/src/core/onKeyDown.ts
+++ b/src/core/onKeyDown.ts
@@ -1,6 +1,6 @@
 import { nextStep, previousStep } from "./steps";
 import exitIntro from "./exitIntro";
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 import isFunction from "../util/isFunction";
 
 /**

--- a/src/core/onResize.ts
+++ b/src/core/onResize.ts
@@ -1,4 +1,4 @@
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 import refresh from "./refresh";
 
 export default function onResize(intro: IntroJs) {

--- a/src/core/placeTooltip.ts
+++ b/src/core/placeTooltip.ts
@@ -5,7 +5,7 @@ import checkRight from "../util/checkRight";
 import checkLeft from "../util/checkLeft";
 import removeEntry from "../util/removeEntry";
 import { HintStep, IntroStep, TooltipPosition } from "./steps";
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 
 /**
  * auto-determine alignment

--- a/src/core/refresh.ts
+++ b/src/core/refresh.ts
@@ -3,7 +3,7 @@ import setHelperLayerPosition from "./setHelperLayerPosition";
 import placeTooltip from "./placeTooltip";
 import fetchIntroSteps from "./fetchIntroSteps";
 import { _recreateBullets, _updateProgressBar } from "./showElement";
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 
 /**
  * Update placement of the intro objects on the screen

--- a/src/core/setHelperLayerPosition.ts
+++ b/src/core/setHelperLayerPosition.ts
@@ -3,7 +3,7 @@ import isFixed from "../util/isFixed";
 import addClass from "../util/addClass";
 import removeClass from "../util/removeClass";
 import setStyle from "../util/setStyle";
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 import { HintStep, IntroStep } from "./steps";
 
 /**

--- a/src/core/showElement.ts
+++ b/src/core/showElement.ts
@@ -11,7 +11,7 @@ import removeShowElement from "./removeShowElement";
 import createElement from "../util/createElement";
 import setStyle from "../util/setStyle";
 import appendChild from "../util/appendChild";
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 import isFunction from "../util/isFunction";
 
 /**

--- a/src/core/steps.ts
+++ b/src/core/steps.ts
@@ -1,7 +1,7 @@
 import isFunction from "../util/isFunction";
 import exitIntro from "./exitIntro";
 import showElement from "./showElement";
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../intro";
 
 export type ScrollTo = "off" | "element" | "tooltip";
 

--- a/src/intro.ts
+++ b/src/intro.ts
@@ -128,7 +128,7 @@ export class IntroJs {
     return this;
   }
 
-  addStep(step: IntroStep) {
+  addStep(step: Partial<IntroStep>) {
     if (!this._options.steps) {
       this._options.steps = [];
     }
@@ -138,7 +138,7 @@ export class IntroJs {
     return this;
   }
 
-  addSteps(steps: IntroStep[]) {
+  addSteps(steps: Partial<IntroStep>[]) {
     if (!steps.length) return this;
 
     for (let index = 0; index < steps.length; index++) {

--- a/src/util/scrollTo.ts
+++ b/src/util/scrollTo.ts
@@ -1,6 +1,6 @@
 import elementInViewport from "./elementInViewport";
 import getWindowSize from "./getWindowSize";
-import { ScrollTo } from "src/core/steps";
+import { ScrollTo } from "../core/steps";
 
 /**
  * To change the scroll of `window` after highlighting an element

--- a/tests/jest/core/dontShowAgain.test.ts
+++ b/tests/jest/core/dontShowAgain.test.ts
@@ -3,7 +3,7 @@ import {
   setDontShowAgain,
   getDontShowAgain,
 } from "../../../src/core/dontShowAgain";
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../../../src/intro";
 
 describe("dontShowAgain", () => {
   test("should call set cookie", () => {

--- a/tests/jest/core/fetchIntroSteps.test.ts
+++ b/tests/jest/core/fetchIntroSteps.test.ts
@@ -1,4 +1,4 @@
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../../../src/intro";
 import fetchIntroSteps from "../../../src/core/fetchIntroSteps";
 
 describe("fetchIntroSteps", () => {

--- a/tests/jest/core/placeTooltip.test.ts
+++ b/tests/jest/core/placeTooltip.test.ts
@@ -1,8 +1,8 @@
 import * as getOffset from "../../../src/util/getOffset";
 import * as getWindowSize from "../../../src/util/getWindowSize";
 import placeTooltip from "../../../src/core/placeTooltip";
-import { IntroJs } from "src/intro";
-import { IntroStep } from "src/core/steps";
+import { IntroJs } from "../../../src/intro";
+import { IntroStep } from "../../../src/core/steps";
 
 describe("placeTooltip", () => {
   test("should automatically place the tooltip position when there is enough space", () => {

--- a/tests/jest/core/steps.test.ts
+++ b/tests/jest/core/steps.test.ts
@@ -1,6 +1,7 @@
 import { nextStep, previousStep } from "../../../src/core/steps";
 import _showElement from "../../../src/core/showElement";
 import { IntroJs } from "../../../src/intro";
+import introJs from "../../../src";
 
 jest.mock("../../../src/core/showElement");
 jest.mock("../../../src/core/exitIntro");
@@ -122,6 +123,40 @@ describe("steps", () => {
 
       expect(fnCompleteCallback).toBeCalledTimes(1);
       expect(fnCompleteCallback).toHaveBeenCalledWith(2, "end");
+    });
+
+    test("should be able to add steps using addStep()", async () => {
+      const intro = introJs();
+
+      intro.addStep({
+        element: document.createElement("div"),
+        intro: "test step",
+      });
+
+      await intro.start();
+
+      expect(intro._introItems).toHaveLength(1);
+      expect(intro._introItems[0].intro).toBe("test step");
+    });
+
+    test("should be able to add steps using addSteps()", async () => {
+      const intro = introJs();
+
+      intro.addSteps([
+        {
+          intro: "first step",
+        },
+        {
+          element: document.createElement("div"),
+          intro: "second step",
+        },
+      ]);
+
+      await intro.start();
+
+      expect(intro._introItems).toHaveLength(2);
+      expect(intro._introItems[0].intro).toBe("first step");
+      expect(intro._introItems[1].intro).toBe("second step");
     });
   });
 });

--- a/tests/jest/core/steps.test.ts
+++ b/tests/jest/core/steps.test.ts
@@ -1,6 +1,6 @@
 import { nextStep, previousStep } from "../../../src/core/steps";
 import _showElement from "../../../src/core/showElement";
-import { IntroJs } from "src/intro";
+import { IntroJs } from "../../../src/intro";
 
 jest.mock("../../../src/core/showElement");
 jest.mock("../../../src/core/exitIntro");

